### PR TITLE
Add Postgres JSON/JSONB support

### PR DESCRIPTION
**Pull request by kak-tus**
Main idea described in readme

* User set an option EnablePostgresJson=true in config (or --enable-postgres-json=true
in command line).
* Gendal generates not Json/Jsonb type for field, but type with same name as field.
* Then user defines this type with SQL interface functions: Scan and Value.
* User can define UnmarshalJSON/MarshalJSON for this struct and simply call then in Scan and Value.
* Also user can override generated type name to some predefined type with UnmarshalJSON/MarshalJSON
and simply add Scan and Value.

### CODE REVIEW

**Comment by turnkey-commerce** [comment]
Hi @kak-tus, thanks for your PR. It looks good and had a few suggestions on the README part. Could you please modify it slightly as below for readability if it looks good to you?

## PostgreSQL JSON/JSONB support
* The user sets an option EnablePostgresJson=true in config (or --enable-postgres-json=true
in command line).
* Gendal does not generate a Json/Jsonb type for the field, but rather a new type with the same name as the field.
* Then the user defines this type with the SQL interface functions: Scan and Value.
* The user can then define UnmarshalJSON/MarshalJSON for this struct and simply call in Scan and Value.
* Also the user can override the generated type name to some predefined type with UnmarshalJSON/MarshalJSON
and simply add Scan and Value.

File: README.md [hunk 1]
```diff
@@ -743,6 +743,15 @@ There is currently no method provided for Oracle as there is no programmatic way
 for which sequences are associated with tables. All PK's will be assumed to be provided
 by the database.
 
+## PostgreSQL JSON/JSONB support
```
Comment (turnkey-commerce) [inline README.md:746]: Some suggestions for minor changes in the README in the comments.

**Comment by turnkey-commerce** [comment]
Never mind, it's probably easier to just accept and then will make the change.

**Comment by kak-tus** [comment]
Sorry for delay. If it is merged - fine!

**Comment by turnkey-commerce** [comment]
No problem, thanks again for the PR!

**Pull request merged**